packages/@sanity/types/src/reference/types.ts

```diff
@@ -2,9 +2,15 @@ import {Path} from '../paths'
 import {SanityDocument} from '../documents'
 
 export interface Reference {
+  _type: string
   _ref: string
   _key?: string
   _weak?: boolean
+  _strengthenOnPublish?: {
+    type: string
+    weak?: boolean
+    template?: {id: string; params: Record<string, string | number | boolean>}
```

Comment:
i think template.params can be any shape because they follow field definitions. [see here](https://www.sanity.io/docs/initial-value-templates#ffd1799becde). might be better as unknown

Reply:
but they are limited to json values, aren't they?